lib/msf/core/opt_string.rb

```diff
@@ -31,7 +31,7 @@ def normalize(value)
   def valid?(value=self.value, check_empty: true)
     value = normalize(value)
     return false if check_empty && empty_required_value?(value)
-    return super
+    return super(check_empty: false)
```

Comment:
This would be best fit in a subclass of `OptString`, as there a number of modules that really shouldn't have empty strings for required arguments. Options that default to the empty string could automatically be converted under the covers to keep code-touch low.

Reply:
I agree here, a new class of `Opt` may be reasonable, maybe even a specific class for `OptPrivate` that is treated like or even shares code with how `private` values are handled for LoginScanners and the like.